src/main/java/org/opensearch/neuralsearch/stats/events/EventStatsManager.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.stats.events;
+
+import lombok.Getter;
+import org.opensearch.neuralsearch.settings.NeuralSearchSettingsAccessor;
+import org.opensearch.neuralsearch.stats.common.StatSnapshot;
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manager class for event counter stats, used to track monotonically increasing counts plus some possible metadata
```

Comment:
```suggestion
 * Manager class for event counter stats, used to track monotonically increasing counts plus some possible metadata
```

Please rewrite to make it more reader friendly. It should provide high level information of this class.